modules/custom/openy_daxko2/src/OpenyActivityFinderDaxkoBackend.php

```diff
@@ -0,0 +1,659 @@
+        foreach ($facets['locations'] as $fl) {
+          if ($fl['id'] == $location['value']) {
+            $locations[$key]['count'] += $fl['count'];
+          }
+        }
+      }
+    }
+
+    return [
+      'count' => $programsResponse['total'],
+      'table' => $result,
+      'pager' => $pager,
+      'facets' => $facets,
+      'groupedLocations' => $locations,
+    ];
+  }
+
+  /**
+   * Get the days of week.
+   */
+  public function getDaysOfWeek() {
+    return [
+      [
+        'label' => 'Monday',
+        'value' => '1',
+      ],
+      [
+        'label' => 'Tuesday',
+        'value' => '2',
+      ],
+      [
+        'label' => 'Wednesday',
+        'value' => '3',
+      ],
+      [
+        'label' => 'Thursday',
+        'value' => '4',
+      ],
+      [
+        'label' => 'Friday',
+        'value' => '5',
+      ],
+      [
+        'label' => 'Saturday',
+        'value' => '6',
+      ],
+      [
+        'label' => 'Sunday',
+        'value' => '7',
+      ],
+    ];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getLocations() {
+    // Need locations groupped to display in filter.
+    return $this->getDaxkoLocations(TRUE);
+  }
+
+  public function getCategories() {
+    // Need locations groupped to display in filter.
+    return $this->getDaxkoCategories(TRUE);
+  }
+
+  public function getCategoriesType() {
+    return 'single';
+  }
+
+  public function getCategoriesTopLevel() {
+    $categories_config = $this->daxkoConfig->get('categories');
+    $categories = $this->parseGrouppedSetting($categories_config, TRUE);
+
+    return array_keys($categories);
+  }
+
+  protected function getDaxkoCategories($group = FALSE) {
+    $categories_config = $this->daxkoConfig->get('categories');
+    $categories = $this->parseGrouppedSetting($categories_config, $group);
+    if (!$group) {
+      return $categories;
+    }
+
+    $numericalKeyedCategories = [];
+    foreach ($categories as $label => $value) {
+      $numericalKeyedCategories[] = ['value' => $value, 'label' => $label];
+    }
+
+    return $numericalKeyedCategories;
+  }
+
+  /**
+   * Run querty to Daxko to get the token.
+   */
+  protected function getDaxkoToken() {
+    $time_start = microtime(true);
+    $client = new Client(['base_uri' => $this->daxkoConfig->get('base_uri')]);
+    $response = $client->request('POST', 'partners/oauth2/token',
+      [
+        'form_params' => [
+          'client_id' => $this->daxkoConfig->get('user'),
+          'client_secret' => $this->daxkoConfig->get('pass'),
+          'grant_type' => 'client_credentials',
+          'scope' => 'client:' . $this->daxkoConfig->get('client_id'),
+        ],
+        'headers' => [
+          'Authorization' => "Bearer " . $this->daxkoConfig->get('referesh_token')
+        ],
+      ]);
+    $time_end = microtime(true);
+    $time = $time_end - $time_start;
+    \Drupal::logger('openy_daxko2')->info('Daxko call. Time %times. URL %url', [
```

Comment:
Use logger channel, passed via DI, instead of calling static methods everywhere https://github.com/ymcatwincities/openy/blob/8.x-2.x/modules/custom/openy_upgrade_tool/openy_upgrade_tool.services.yml#L2